src/serac/physics/state/state_manager.cpp

```diff
@@ -76,30 +81,50 @@ void StateManager::initialize(axom::sidre::DataStore& ds, const std::string& out
   }
 }
 
-FiniteElementState StateManager::newState(FiniteElementVector::Options&& options, const std::string& mesh_tag)
+FiniteElementState& StateManager::shapeDisplacement(const std::string& mesh_tag)
+{
+  return *shape_displacements_[mesh_tag];
+}
+
+FiniteElementDual& StateManager::shapeDisplacementSensitivity(const std::string& mesh_tag)
+{
+  return *shape_sensitivities_[mesh_tag];
+}
+
+void StateManager::storeState(FiniteElementState& state)
 {
   SLIC_ERROR_ROOT_IF(!ds_, "Serac's data store was not initialized - call StateManager::initialize first");
-  SLIC_ERROR_ROOT_IF(datacolls_.find(mesh_tag) == datacolls_.end(),
-                     axom::fmt::format("Mesh tag \"{}\" not found in the data store", mesh_tag));
-  SLIC_ERROR_ROOT_IF(named_states_.find(options.name) != named_states_.end(),
-                     axom::fmt::format("StateManager already contains a state named {}", options.name));
+  auto mesh_tag = collectionID(&state.mesh());
+  SLIC_ERROR_ROOT_IF(named_states_.find(state.name()) != named_states_.end(),
+                     axom::fmt::format("StateManager already contains a state named {}", state.name()));
   auto&                  datacoll = datacolls_.at(mesh_tag);
-  const std::string      name     = options.name;
-  auto                   state    = FiniteElementState(mesh(mesh_tag), std::move(options));
+  const std::string      name     = state.name();
   mfem::ParGridFunction* grid_function;
   if (is_restart_) {
     grid_function = datacoll.GetParField(name);
     state.setFromGridFunction(*grid_function);
   } else {
     SLIC_ERROR_ROOT_IF(datacoll.HasField(name),
-                       axom::fmt::format("Serac's datacollection was already given a field named '{0}'", name));
+                       axom::fmt::format("StateManager already given a field named '{0}'", name));
 
     // Create a new grid function with unallocated data. This will be managed by sidre.
     grid_function = new mfem::ParGridFunction(&state.space(), static_cast<double*>(nullptr));
     datacoll.RegisterField(name, grid_function);
     state.setFromGridFunction(*grid_function);
   }
   named_states_[name] = grid_function;
+}
+
+FiniteElementState StateManager::newState(FiniteElementVector::Options&& options, const std::string& mesh_tag)
+{
+  SLIC_ERROR_ROOT_IF(!ds_, "Serac's data store was not initialized - call StateManager::initialize first");
+  SLIC_ERROR_ROOT_IF(datacolls_.find(mesh_tag) == datacolls_.end(),
+                     axom::fmt::format("Mesh tag \"{}\" not found in the data store", mesh_tag));
```

Comment:
We, not specifically this PR, really are all over the place quoting injected values... sigh.

Reply:
Is this saying you would prefer to not put quotes around them? It seems appropriate here, since it makes clear the exact value of the `mesh_tag` string.

Reply:
I prefer to quote the values that are given in error messages.

Example:
```
Error: The value 'foo' given by user is super duper bad.
```

Reply:
I've made changes to most of the formatted strings in serac (I skipped a few in serac/infrastructure). I use single quotes around the substitutions to avoid having to escape the double quote character in the string. For some substitutions with numeric output (rather than indices, strings)

```cpp
axom::fmt::format("Bulk sensitivity vector norm: {}", bulk_norm)
```

I just left the single quotes off, as it seems to produce cleaner output.